scheduler/src/modules/transactions/transaction.service.ts

```diff
@@ -0,0 +1,112 @@
+  private holidays: string[] = [];
+  constructor(
+    @InjectRepository(Transaction)
+    private readonly transactionRepository: Repository<Transaction>,
+    @InjectRepository(TransactionDetail)
+    private readonly cfsTransactionDetailRepo: Repository<TransactionDetail>,
+    @InjectRepository(BcHoliday)
+    private readonly holidayRepository: Repository<BcHoliday>,
+  ) {
+    void this.initializeHolidays();
+  }
+
+  private async initializeHolidays(): Promise<void> {
+    this.holidays = await this.getBcHolidays();
+  }
+
+  async getAllTransactions(): Promise<Transaction[]> {
+    return this.transactionRepository.find();
+  }
+
+  async getHolidays(): Promise<BcHoliday[]> {
+    const now: Date = new Date();
+    const year = now.getFullYear().toString(); // Convert year to string
+    
+    return this.holidayRepository.find({
+      where: { holidayDate: Like(`%${year}%`) }
+    });
+  }
+
+  async getTransactionDetails(): Promise<Transaction[]> {
+    const transactions = await this.transactionRepository.createQueryBuilder('transaction')
+    .innerJoinAndSelect('TransactionDetail', 'detail', 'transaction.TRANSACTION_ID = detail.TRANSACTION_ID')
+    .where('detail.CFS_FILE_STATUS_TYPE = :status', { status: 'READY' })
+    .getMany();
+
+    if (transactions.length > 0) {
+      console.log(transactions);
+      return transactions;
+    } else {
+      console.log('No transactions found with status READY');
+      return [];
+    }
+  }
+
+  async updateCfsFileStatusType(fileName: string): Promise<void> {
+    const currentDate = new Date();
+    const currentUTCTimestamp = currentDate.toISOString();
+    const transactionDetails: TransactionDetail[] = await this.cfsTransactionDetailRepo.find({ where: { CFS_FILE_STATUS_TYPE: 'READY' } });
+    const updatedTransactionDetails: TransactionDetail[] = transactionDetails.map(detail => {
+      detail.CFS_FILE_STATUS_TYPE = 'SENT';
+      detail.PROCESSSING_DATE_TIME = currentUTCTimestamp;
+      detail.FILE_NAME = fileName;
+      return detail;
+    });
+
+    try {
+      await this.cfsTransactionDetailRepo.save(updatedTransactionDetails);
+      console.log(`Updated transaction details`);
+    } catch (error) {
+      console.error('Error updating transaction details:', error);
+      throw error;
+    }
+  }
+
+  private async getBcHolidays(): Promise<string[]> {
+    try {
+        const holidays: BcHoliday[] = await this.getHolidays();
+
+        const formatHoliday = (holiday: BcHoliday): string => {
+          return holiday.holidayDate;
+        };
+        
+        const holidayStrings: string[] = holidays.map(formatHoliday);
+        return holidayStrings;
+
+    } catch (error) {
+        console.error('Error fetching BC holidays:', error);
+        return [];
+    }
+  }
+
+  private isHoliday(date: Date): boolean {
+    const formattedDate = date.toISOString().split('T')[0];
+    return this.holidays.includes(formattedDate);
+  }
+
+  @Cron('30 16 * * 1-5')
```

Comment:
Although seconds are optional in cron pattern, just for consistency add 6 places for cron pattern , we can use a leading 0 to represent seconds.